### ISSUE TITLE
An idea and one bug

### ISSUE THREAD

**Issue by foxya27** [enjarai/headpats#2]
When I was petted by players on a server with the Purpur core, after they stopped doing this, I continued to purr and my head was shaking by itself. For some reason, only I have this problem, and others see from the outside that my head is shaking, I suggest adding a check to the mod.
Idea: Add fox snorting, this is the perfect theme for my server.
Add the ability to change shaking and sounds without the YACL mod.
And please send me another localization file, I’ll translate it to RU

**Comment by foxya27** [comment]
![ezgif-3-0059d8ad1e](https://github.com/user-attachments/assets/c76f5c11-fbb1-4231-8741-23f2fab71fe4)

**Comment by enjarai** [comment]
I guess this is a desync issue with the plugin... So your head keeps shaking for all players, not just yourself? Can you make it stop by getting pet again, or is it permanently stuck?

I can add some more config options for sure, when do you imagine the fox snort sound playing? The purr is easier because its continuous...

The language file in the mod is located here, though I suggest you wait a bit until I add more config options:
https://github.com/enjarai/headpats/blob/master/fabric/src/main/resources/assets/headpats/lang/en_us.yml

**Comment by foxya27** [comment]
We found how to reproduce when 2 or more people stroke you for 30 seconds, this bug occurs and does not go away

**Comment by foxya27** [comment]
Re-entering the game resets the bug until they patted  you again

**Comment by enjarai** [comment]
Oh so it has to be two or more? That gives me something to look into at least...

**Comment by foxya27** [comment]
> Oh so it has to be two or more? That gives me something to look into at least...

Yes, they decided to pet me with the entire server for a minute, and that’s how we found this bug

**Comment by foxya27** [comment]
> I guess this is a desync issue with the plugin... So your head keeps shaking for all players, not just yourself? Can you make it stop by getting pet again, or is it permanently stuck?
> 
> I can add some more config options for sure, when do you imagine the fox snort sound playing? The purr is easier because its continuous...
> 
> The language file in the mod is located here, though I suggest you wait a bit until I add more config options: https://github.com/enjarai/headpats/blob/master/fabric/src/main/resources/assets/headpats/lang/en_us.yml

As soon as you release an update with sounds, I will transfer it to ru_ru

**Comment by foxya27** [comment]
Is any more information needed to fix this?

**Comment by foxya27** [comment]
And yes, he doesn’t stop if they patting  me again, and the only thing that helps is re-logging into the server, and then before the first patting

**Comment by foxya27** [comment]
Players ask you to add a whitelist of those who can pet you or simply turn it off in the mod settings

**Comment by enjarai** [comment]
> Is any more information needed to fix this?

I'm honestly not sure, but I'll try some things. Would I be able to give you a plugin jar to test? I don't have as many players available, and I havent had it happen with 3.

**Comment by foxya27** [comment]
> > Is any more information needed to fix this?
> 
> I'm honestly not sure, but I'll try some things. Would I be able to give you a plugin jar to test? I don't have as many players available, and I havent had it happen with 3.

Yes, let me reinstall the plugin

**Comment by enjarai** [comment]
Would you be able to join my Discord? That'd make real-time communication a little easier.

**Comment by foxya27** [comment]
I will wait .jar

**Comment by foxya27** [comment]
![image](https://github.com/user-attachments/assets/99e453ed-8c2d-425c-ac3b-41db120969f1)
here?

> Would you be able to join my Discord? That'd make real-time communication a little easier.
